### ISSUE TITLE
Location updates sometimes way off (100+ km)

### ISSUE THREAD

**Issue by rs443** [home-assistant/android#929]
**Home Assistant Android version:**
2.4.0-full

**Android version:**
11

**Phone model:**
Google Pixel 3a

**Home Assistant version:**
0.114.4

**Last working Home Assistant release (if known):**
2.3.3-full

**Description of problem:**
Location updates is sometimes way off. Today I noticed that Home Assistant reported me in a location about 110 kilometers away. According to the logs HA was convinced that I was located there with several location updates with good accuracy during a period of an hour.

The strange location was set at 09:03, and reset to my real location at 09:54.

The only thing that stands out is that altitude, course and vertical_accuracy is all 0.  Can it be some sort of IP geolocation update? 

**Traceback (if applicable):**
```
2020-09-14 06:31:44 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload for type update_location: {'location_name': '', 'gps': [N.298841, E.206636], 'gps_accuracy': 3, 'speed': 0, 'altitude': 26, 'course': 257, 'vertical_accuracy': 0}
2020-09-14 06:39:53 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload for type update_location: {'location_name': '', 'gps': [N.298833, E.2066507], 'gps_accuracy': 3, 'speed': 0, 'altitude': 26, 'course': 96, 'vertical_accuracy': 0}
2020-09-14 07:03:11 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload for type update_location: {'location_name': '', 'gps': [N.6455586, E.0787489], 'gps_accuracy': 56, 'speed': 0, 'altitude': 0, 'course': 0, 'vertical_accuracy': 0}  << started here
2020-09-14 07:15:04 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload for type update_location: {'location_name': '', 'gps': [N.6455563, E.078735], 'gps_accuracy': 104, 'speed': 0, 'altitude': 0, 'course': 0, 'vertical_accuracy': 0}
2020-09-14 07:23:08 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload for type update_location: {'location_name': '', 'gps': [N.645557, E.0787459], 'gps_accuracy': 68, 'speed': 0, 'altitude': 0, 'course': 0, 'vertical_accuracy': 0}
2020-09-14 07:23:08 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload for type update_location: {'location_name': '', 'gps': [N.645557, E.0787459], 'gps_accuracy': 68, 'speed': 0, 'altitude': 0, 'course': 0, 'vertical_accuracy': 0}
2020-09-14 07:23:08 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload for type update_location: {'location_name': '', 'gps': [N.6455596, E.0787498], 'gps_accuracy': 45, 'speed': 0, 'altitude': 0, 'course': 0, 'vertical_accuracy': 0}
2020-09-14 07:38:01 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload for type update_location: {'location_name': '', 'gps': [N.6455588, E.0787488], 'gps_accuracy': 32, 'speed': 0, 'altitude': 0, 'course': 0, 'vertical_accuracy': 0}
2020-09-14 07:50:15 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload for type update_location: {'location_name': '', 'gps': [N.6455517, E.0787404], 'gps_accuracy': 116, 'speed': 0, 'altitude': 0, 'course': 0, 'vertical_accuracy': 0}
2020-09-14 07:50:15 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload for type update_location: {'location_name': '', 'gps': [N.6455538, E.0787415], 'gps_accuracy': 122, 'speed': 0, 'altitude': 0, 'course': 0, 'vertical_accuracy': 0}
2020-09-14 07:54:16 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload for type update_location: {'location_name': '', 'gps': [N.6455568, E.0787363], 'gps_accuracy': 92, 'speed': 0, 'altitude': 0, 'course': 0, 'vertical_accuracy': 0}
2020-09-14 07:54:16 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload for type update_location: {'location_name': '', 'gps': [N.2989235, E.2072265], 'gps_accuracy': 6, 'speed': 0, 'altitude': 41, 'course': 26, 'vertical_accuracy': 12}
```

**Screenshot of problem:**
![100km](https://user-images.githubusercontent.com/46565725/93064086-6b4d7180-f677-11ea-8e61-a8e6c41a666a.PNG)

**Additional information:**

- App is excluded from battery optimizations and data management.
- There are no apps running in background except Home assistant, Opera browser and Spotify.
- Looking up my Google Maps location timeline it never has reported me there.
- Background location & Zone Location: Enabled.
- Android Wi-Fi and Bluetooth location scanning: Disabled

**Comment by rs443** [comment]
After some investigation i suspect that this has to do with Wi-Fi location scanning. At work there are several Wifi APs and one of them is probably has some geolocation information stored. Or maybe one of the SSIDs are present at multiple locations and cities.

Wi-Fi location scanning is disabled on the phone, does the HA companion use it anyway?

**Comment by JBassett** [comment]
We use the [Fused Location Provider](https://developers.google.com/location-context/fused-location-provider) which combines GPS, cell tower, and wifi to best estimate your location.

**Comment by anyuta1166** [comment]
> Wi-Fi location scanning is disabled on the phone

Wi-Fi location scanning setting is effective only when Wi-Fi is off.

When Wi-Fi is on, scanning is always enabled.

**Comment by blair287** [comment]
I'm having this same issue always home then away the. Home again and vice versa in the space on a few seconds.

![Screenshot_20200927-065441](https://user-images.githubusercontent.com/44187409/94357175-61236e00-008e-11eb-907f-6aad232400ef.jpg)

The times it happened is within a few seconds of the first update which is always correct.

Is it not possible to add in a denounce accept the first result then stop further update for say 30 seconds.

It's screwing up automations with this jumping back and forth sometimes in will go home away home away home all within 60 seconds and my home zone is 400m.

**Comment by dshokouhi** [comment]
@rs443 I think we may be good to close this one too? On hindsight it may have been a duplicate of #871 based on the fix in #1119

**Comment by rs443** [comment]
@dshokouhi I agree! Also I have not seen this issue for a while.
